selfdrive/car/car_specific.py

```diff
@@ -0,0 +1,239 @@
+        if CS.low_speed_lockout:
+          events.add(EventName.lowSpeedLockout)
+        if CS.out.vEgo < self.CP.minEnableSpeed:
+          events.add(EventName.belowEngageSpeed)
+          if CC_prev.actuators.accel > 0.3:
+            # some margin on the actuator to not false trigger cancellation while stopping
+            events.add(EventName.speedTooLow)
+          if CS.out.vEgo < 0.001:
+            # while in standstill, send a user alert
+            events.add(EventName.manualRestart)
+
+    elif self.CP.carName == 'gm':
+      # The ECM allows enabling on falling edge of set, but only rising edge of resume
+      events = self.create_common_events(CS, CS_prev, extra_gears=[GearShifter.sport, GearShifter.low,
+                                                                   GearShifter.eco, GearShifter.manumatic],
+                                         pcm_enable=self.CP.pcmCruise, enable_buttons=(ButtonType.decelCruise,))
+      if not self.CP.pcmCruise:
+        if any(b.type == ButtonType.accelCruise and b.pressed for b in CS.out.buttonEvents):
+          events.add(EventName.buttonEnable)
+
+      # Enabling at a standstill with brake is allowed
+      # TODO: verify 17 Volt can enable for the first time at a stop and allow for all GMs
+      below_min_enable_speed = CS.out.vEgo < self.CP.minEnableSpeed or CS.moving_backward
+      if below_min_enable_speed and not (CS.out.standstill and CS.out.brake >= 20 and
+                                         self.CP.networkLocation == NetworkLocation.fwdCamera):
+        events.add(EventName.belowEngageSpeed)
+      if CS.out.cruiseState.standstill:
+        events.add(EventName.resumeRequired)
+      if CS.out.vEgo < self.CP.minSteerSpeed:
+        events.add(EventName.belowSteerSpeed)
+
+    elif self.CP.carName == 'volkswagen':
+      events = self.create_common_events(CS, CS_prev, extra_gears=[GearShifter.eco, GearShifter.sport, GearShifter.manumatic],
+                                         pcm_enable=not self.CP.openpilotLongitudinalControl,
+                                         enable_buttons=(ButtonType.setCruise, ButtonType.resumeCruise))
+
+      # Low speed steer alert hysteresis logic
+      if (self.CP.minSteerSpeed - 1e-3) > VWCarControllerParams.DEFAULT_MIN_STEER_SPEED and CS.out.vEgo < (self.CP.minSteerSpeed + 1.):
+        self.low_speed_alert = True
+      elif CS.out.vEgo > (self.CP.minSteerSpeed + 2.):
+        self.low_speed_alert = False
+      if self.low_speed_alert:
+        events.add(EventName.belowSteerSpeed)
+
+      if self.CP.openpilotLongitudinalControl:
+        if CS.out.vEgo < self.CP.minEnableSpeed + 0.5:
+          events.add(EventName.belowEngageSpeed)
+        if CC_prev.enabled and CS.out.vEgo < self.CP.minEnableSpeed:
+          events.add(EventName.speedTooLow)
+
+      # TODO: we don't have the CC here
+      # if self.CC.eps_timer_soft_disable_alert:
+      #   events.add(EventName.steerTimeLimit)
```

Comment:
@jyoung8607 does it make sense to move this into the car interface, the way other brands use it to store temporary variables not going in carstate? Eventually we will want to standardize everything into the CarState struct anyway